dlt/common/destination/reference.py

```diff
@@ -49,6 +51,39 @@ def __init__(
         ) -> None:
             ...
 
+@configspec(init=True)
+class DestinationClientStagingConfiguration(DestinationClientDwhConfiguration):
+    as_staging: bool = False
```

Comment:
I really think that `as_staging` is the only flag we need (if we do not embed this fully into DestinationClientDwhConfiguration)